### ISSUE TITLE
Allow lua 5.4 import

### ISSUE THREAD

**Issue by SlashScreen** [pigpigyyy/Yuescript#156]
Lua 5.4 uses `import` instead of `require`. Is there a way we can use `import "module"` instead of `require("module")` in generated code?

**Comment by pigpigyyy** [comment]
I can not find the import feature you mentioned in official doc  http://www.lua.org/manual/5.4/. Can you provide a link or something to get me informed?

**Comment by SlashScreen** [comment]
Fair point, perhaps I was mistaken. Playdate's embedded Lua only uses the `import` keyword, not `require`. Right now I've been trying to do this with macros, but since Playdate doesn't have `require`, I cannot use `from "class" import A`. Would it be possible to add a toggle or some kind of config? 
Also, is `import "a/b"` supposed to result in a syntax error?

**Comment by pigpigyyy** [comment]
"import" is treated as a keyword for the moment, but I can try to change that. And supporting 'import("module")' as a function call.

**Comment by chrsm** [comment]
Would a macro suffice here?

```moonscript
export macro raw = (code) ->
  { :code, type: "lua" }

-- usage

$raw [[ import "x" ]]
```

This is what I do since `xmake` uses an `import` function:

```moonscript
    $raw[[
      import("core.base.option")
    ]]
```

**Comment by SlashScreen** [comment]
@chrsm Macros would work, but what about import from and import as?

**Comment by SkyyySi** [comment]
I'd say that `import` should be turned into a soft keyword that's only reserved when used as a statement, but not as an expression, so that

```moonscript
import "foo"
```

still compiles to

```lua
local foo = require("foo")
```

but

```moonscript
foo = import "foo"
```

becomes

```lua
local foo = import("foo")
```

---

If someone wanted to make the `import` statement work in an environment that renames `require` to `import`, they could simple put something like

```moonscript
global require = import
```

at the beginning of their script.

**Comment by pigpigyyy** [comment]
Digging around the parser rules a little while, I'm thinking of adding special aliasing syntax like:
```moonscript
`import "abc" -- the backtick will turn keyword import to variable name
```
Or we have to keep using macro function like this to get similar result:
```moonscript
macro import = (moduleName) -> "_G['import'](#{moduleName})"

$import "abc.zyx"
mod = $import "module"

nil
```
compiles to:
```lua
_G['import']("abc.zyx")
local mod = _G['import']("module")
return nil
```

**Comment by pigpigyyy** [comment]
@SlashScreen 
> Fair point, perhaps I was mistaken. Playdate's embedded Lua only uses the `import` keyword, not `require`. Right now I've been trying to do this with macros, but since Playdate doesn't have `require`, I cannot use `from "class" import A`. Would it be possible to add a toggle or some kind of config? Also, is `import "a/b"` supposed to result in a syntax error?

If you want to use the `import()` function with `from "class" import A` syntax in current Yuescript. We can do it with:
```moonscript
macro Playdate = (moduleName) -> "_G.import(#{moduleName})"

from $Playdate("module") import a, b, c
```
get
```lua
local a, b, c
do
  local _obj_0 = _G.import("module")
  a, b, c = _obj_0.a, _obj_0.b, _obj_0.c
end
```

`import "a/b"` is supposed to result in a syntax error. You have to do `import "a.b"` as a Lua common practice instead. Or use `require "a/b"`.
